epam_tr/src/main/java/com/epam/tr/service/UserServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.epam.tr.service;
+
+import com.epam.tr.dao.Dao;
+import com.epam.tr.dto.UserDto;
+import com.epam.tr.entities.AppUser;
+import com.epam.tr.entities.UserRole;
+import com.epam.tr.exceptions.InvalidCredentialsException;
+import com.epam.tr.service.logic.validator.Validator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private Dao<AppUser> dao;
+    private Validator<AppUser> validator;
+
+    @Autowired
+    public UserServiceImpl(Dao<AppUser> dao, Validator<AppUser> validator) {
+        this.dao = dao;
+        this.validator = validator;
+    }
+
+    @Override
+    public void create(AppUser entity) throws InvalidCredentialsException {
+        validator.isValid(entity);
+        dao.insert(entity);
+    }
+
+    @Override
+    public void update(int oldUserId, UserDto newEntity) throws InvalidCredentialsException {
+        validator.isValid(newEntity.to());
```

Comment:
to is unobvious name.
in this case it makes sense to name like toUser.
Yes, I know that I mentioned "to".